ui/src/data_explorer/components/WriteDataForm.js

```diff
@@ -0,0 +1,118 @@
+import React, {Component, PropTypes} from 'react'
+
+import DatabaseDropdown from 'shared/components/DatabaseDropdown'
+import OnClickOutside from 'shared/components/OnClickOutside'
+
+class WriteDataForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      selectedDatabase: null,
+      inputContent: '',
+    }
+
+    this.handleSelectDatabase = ::this.handleSelectDatabase
+    this.handleSubmit = ::this.handleSubmit
+    this.handleClickOutside = ::this.handleClickOutside
+    this.handleKeyUp = ::this.handleKeyUp
+    this.handleEdit = ::this.handleEdit
+  }
+
+  handleSelectDatabase(item) {
+    this.setState({selectedDatabase: item.text})
+  }
+
+  handleClickOutside(e) {
+    // guard against clicking to close error notification
+    if (e.target.className === 'overlay-technology') {
```

Comment:
There may be some possible solution using `ref`. What you could do, to make it less brittle, is to factor out the `'overlay-technology'` string here and wherever it's used in OT as a constant, and import that constant into both files, so that they are coupled.